### ISSUE TITLE
Add `IGNORE` QueueNotFoundStrategy Option for SQS

### ISSUE THREAD

**Issue by OmarTareefy** [awspring/spring-cloud-aws#1143]
**Type**: Feature

**Is your feature request related to a problem? Please describe.**
Yes, currently in SqsMessageListenerContainerFactory there are two strategies to consider when a queue is not found, either CREATE or FAIL. In our business use case the CREATE will not be the best for us because we create the AWS queue with certain configurations and permissions and also we can't give the container running the application a **create-queue** permissions, in the other hand the FAIL strategy will not be good for us because in some cases the queue might not be created for that specific deployment and we would want the application to just start even if the queue is not there.

**Describe the solution you'd like**
It will be great if a new strategy is created called DO_NOTHING in QueueNotFoundStrategy enum where spring just ignores initializing the SqsMessageListenerContainerFactory for that SqsListener because the queue doesn't exist and the application start without issues.

**Describe alternatives you've considered**
I tagged my SQSConfig @Configuration class with @ConditionalOnExpression(property == true) (this will be true by default) and created a spring bootstrap configuration class that pings the related queue using getQueueUrl and if that throws an exception (which means the queue is not there or the application doesn't have permissions) I set that property to false so that the SQSConfig @Configuration class is turned off,

Another solution we considered is to set the autoStartup to false in the defaultSqsListenerContainerFactory and then at some point in the app startup, we get all of the containers from MessageListenerContainerRegistry and start them one by one, if the starting fails (because the queue is not created) we simply log an error and the app continues.

Please let me know if there is a better workaround or if there is a better solution for the issue I am facing that I am not aware of,
Thank you!

**Comment by tomazfernandes** [comment]
Hey @OmarTareefy, your workarounds sound reasonable.

Do they work well for you?

I think the main challenge for implementing this in the framework would be figuring out how to graciously handle the error thrown by the `QueueAttributesResolver` and interrupting container startup, but it should be doable.

PRs are welcome!

**Comment by OmarTareefy** [comment]
Yes, I ended up setting the auto start to false then added a logic when our application starts to start those containers and gracefully handled the errors if the queue is not there

`try {
            MessageListenerContainerRegistry messageListenerContainerRegistry =
                    ApplicationContextProvider.getBean(MessageListenerContainerRegistry.class);
            messageListenerContainerRegistry
                    .getListenerContainers()
                    .forEach(messageListenerContainer -> {
                SqsMessageListenerContainer sqsMessageListenerContainer = null;
                try {
                    if (messageListenerContainer instanceof SqsMessageListenerContainer && !messageListenerContainer.isRunning()) {
                        sqsMessageListenerContainer = (SqsMessageListenerContainer) messageListenerContainer;
                        sqsMessageListenerContainer.start();
                        Log.info(this, "Message Listener Container was started successfully for queue: " + sqsMessageListenerContainer.getQueueNames());
                    }
                } catch (Exception e) {
                    Log.error(this, String.format("Unable to start Message Listener Container for queue: %s due to %s",
                            sqsMessageListenerContainer.getQueueNames(), e.getMessage()));
                }
            });
        } catch (Exception e) {
            Log.error(this, e.getMessage(), e);
        }`

**Comment by shreyas957** [comment]
Hi, @tomazfernandes, I'm happy to work on this, could you please explain bit more on how we can achieve this?? 
I think we can return empty queueUrl and we stop container .
we have already stop() method for the same .
```java
@Override
	protected void doStart() {
		Assert.notNull(this.sqsAsyncClient, "sqsAsyncClient not set");
		Assert.notNull(this.queueAttributeNames, "queueAttributeNames not set");
		this.queueAttributes = resolveQueueAttributes();
		this.queueUrl = this.queueAttributes.getQueueUrl();
		if (this.queueNotFoundStrategy == QueueNotFoundStrategy.IGNORE && this.queueUrl.isEmpty()) {
			logger.warn("Queue {} not found and IGNORE strategy is used. Container will not poll messages.", this.queueAttributes.getQueueName());
			stop();
			return;
		}
		configureConversionContextAndAcknowledgement();
	}
```

Please le me know. 
I am happy to learn from you and do it.

**Comment by shreyas957** [comment]
Any update on this? @tomazfernandes  I am happy to work on this, can you share more info about how we can do this.

**Comment by alextadams88** [comment]
Seconding this request, especially because this was the default behavior on the older `spring-cloud-starter-aws-messaging` library for Spring Cloud AWS 2 - it would simply ignore queues that didn't exist until you tried to write to the queue. 

So when upgrading Spring Cloud AWS versions from 2 to 3, we have to refactor code that depended on this functionality.

**Comment by tomazfernandes** [comment]
Hey folks - I think that's a valid use case.

As I mentioned:
> I think the main challenge for implementing this in the framework would be figuring out how to graciously handle the error thrown by the QueueAttributesResolver and interrupting container startup, but it should be doable.

I don't have a solution in mind, but happy to take a look at a PR with a proposal.

**Comment by fatih-onventis** [comment]
I've hit this behavioral change during the dependency upgrade to v3 as well. I wasn't expecting to see a change like this at least without having a similar option tbh.

**Comment by BK202503** [comment]
Opened #1640 — adds `IGNORE` as a third `QueueNotFoundStrategy`. Wires through a new `QueueNotFoundException` (subtype of `QueueAttributesResolvingException`) so `AbstractSqsMessageSource` can recognize it, skip startup, and return empty polls; `FAIL` / `CREATE` paths untouched. Integration test against LocalStack included.
